mobile/mobile-core/android-process-service.ts

```diff
@@ -51,46 +51,76 @@ export class AndroidProcessService implements Mobile.IAndroidProcessService {
 		}).future<string>()();
 	}
 
+	public getMappedAbstractToTcpPorts(deviceIdentifier: string, appIdentifiers: string[]): IFuture<IDictionary<number>> {
+		return ((): IDictionary<number> => {
+			let adb = this.getAdb(deviceIdentifier),
+				abstractPortsInformation = this.getAbstractPortsInformation(adb).wait(),
+				processIds = this.getProcessIds(adb, appIdentifiers).wait(),
+				adbForwardList = adb.executeCommand(["forward", "--list"]).wait(),
+				localPorts: IDictionary<number> = {};
+
+			_.each(appIdentifiers, appIdentifier => {
+				localPorts[appIdentifier] = null;
+				let processId = processIds[appIdentifier];
+
+				if (!processId) {
+					return;
+				}
+
+				let abstractPort = this.getAbstractPortForApplication(adb, processId, appIdentifier, abstractPortsInformation).wait();
+
+				if (!abstractPort) {
+					return;
+				}
+
+				let localPort = this.getAlreadyMappedPort(adb, deviceIdentifier, abstractPort, adbForwardList).wait();
+
+				if (localPort) {
+					localPorts[appIdentifier] = localPort;
+				}
+			});
+
+			return localPorts;
+		}).future<IDictionary<number>>()();
+	}
+
 	public getDebuggableApps(deviceIdentifier: string): IFuture<Mobile.IDeviceApplicationInformation[]> {
 		return ((): Mobile.IDeviceApplicationInformation[] => {
 			let adb = this.getAdb(deviceIdentifier);
 			let androidWebViewPortInformation = (<string>this.getAbstractPortsInformation(adb).wait()).split(EOL);
 
 			// TODO: Add tests and make sure only unique names are returned.
 			return _(androidWebViewPortInformation)
-				.map((line: string) => this.getApplicationInfoFromWebViewPortInformation(adb, deviceIdentifier, line).wait())
-				.filter(appIdentifier => !!appIdentifier)
-				.uniqBy("appIdentifier")
+				.map((line: string) => {
```

Comment:
You probably don't need the `( )` brackets and you definitely don't need the `{ }` ones